### PR TITLE
Rhel 9 Migration for 3.x Branch

### PR DESCRIPTION
**Pull request by SDawley**
https://issues.redhat.com/browse/CRW-3185
Changing everything that pointed to rhel 8 to point to rhel 9.
I expect some things like idea will need to be changed...

### CODE REVIEW

File: devspaces-operator/editors-definitions/che-idea.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ events:
   postStart:
     - init-che-idea-command
 components:
-  - name: idea-rhel8
```
Comment (svor) [inline devspaces-operator/editors-definitions/che-idea.yaml:78]: The image is ubi8 based, look at line 80. Should we change the image or continue using idea-rhel8 name (line 78 and 69)
cc @azatsarynnyy

File: devspaces-pluginregistry/README.md [hunk 2]
```diff
@@ -11,17 +11,17 @@ Luckily there is an automatic way to build and publish .vsix files and their sou
 ## Plugin Registry Build Process
 The DS plugin registry is a fork of the upstream [Che plugin registry](https://github.com/eclipse-che/che-plugin-registry). While the build logic to build the actual registry is very similar to upstream, the steps required to actually build + publish the DS plugin registry images is quite different.
 
-To build the devfile registry for use with a specific Dev Spaces 3.yy version, it is necessary to check out the  appropriate `devspaces-3.yy-rhel-8` branch. This will ensure the build contains the correct tags and references to the images. The `devspaces-3-rhel-8` branch is for the upcoming 3.next release.
```
Comment (svor) [inline devspaces-pluginregistry/README.md:14]: To build the **plugin** registry

Comment (svor) [inline devspaces-pluginregistry/README.md:14]: To build the **plugin** registry

File: devspaces-pluginregistry/build.sh [hunk 3]
```diff
@@ -151,8 +151,8 @@ prepareOpenvsxPackagingAsset() {
     fi
 
     SCRIPT_BRANCH="$(git rev-parse --abbrev-ref HEAD 2>/dev/null || true)"
-    if [[ $SCRIPT_BRANCH != "devspaces-3."*"-rhel-8" ]]; then
```
Comment (svor) [inline devspaces-pluginregistry/build.sh:154]: would not these changes be rewritten when synced with the upstream version: https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/build.sh ?

File: devspaces-pluginregistry/build/dockerfiles/Dockerfile [hunk 4]
```diff
@@ -130,7 +130,7 @@ RUN \
     chgrp -R 0 /var/lib/pgsql /var/lib/pgsql/15 /var/lib/pgsql/data /var/lib/pgsql/backups && \
     mv /var/lib/pgsql/15/data/database /var/lib/pgsql/15/data/old
 
-ARG DS_BRANCH=devspaces-3-rhel-8
```
Comment (svor) [inline devspaces-pluginregistry/build/dockerfiles/Dockerfile:133]: would not these changes be rewritten when synced with the upstream version: https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/build/dockerfiles/Dockerfile#L133 ?

File: devspaces-pluginregistry/build/dockerfiles/rhel.Dockerfile [hunk 5]
```diff
@@ -134,7 +134,7 @@ RUN \
     chgrp -R 0 /var/lib/pgsql /var/lib/pgsql/15 /var/lib/pgsql/data /var/lib/pgsql/backups && \
     mv /var/lib/pgsql/15/data/database /var/lib/pgsql/15/data/old
 
-ARG DS_BRANCH=devspaces-3-rhel-8
```
Comment (svor) [inline devspaces-pluginregistry/build/dockerfiles/rhel.Dockerfile:137]: would not these changes be rewritten when synced with the upstream version: https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/build/dockerfiles/rhel.Dockerfile#L137 ?

File: devspaces-pluginregistry/build/scripts/download_vsix.sh [hunk 6]
```diff
@@ -8,7 +8,7 @@ set -o pipefail
 openvsxJson="/openvsx-server/openvsx-sync.json"
 
 usage() {
-    echo "Usage: $0 -b devspaces-3.y-rhel-8 -j /path/to/openvsx-sync.json --no-download
```
Comment (svor) [inline devspaces-pluginregistry/build/scripts/download_vsix.sh:11]: would not these changes be rewritten when synced with the upstream version: https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/build/scripts/download_vsix.sh#L11

**Comment by svor** [comment]
Are we going to have rhel-9 branches in all samples repos from this list https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/product/tagRelease.sh#L252-L272. If yes, we need to update samples URLs here: https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-dashboard/samples/index.json#L11

**Comment by SDawley** [comment]
Updated the index.json.

**Review by svor — APPROVED** [review]
lgtm

File: devspaces-operator/editors-definitions/che-idea.yaml (hunk 1 above)
Comment (SDawley) [reply devspaces-operator/editors-definitions/che-idea.yaml]: For now I'm going to leave idea as rhel8.

**Pull request merged**